diff --git a/src/flask/cli.py b/src/flask/cli.py
index d85385b..f8373d0 100644
--- a/src/flask/cli.py
+++ b/src/flask/cli.py
@@ -124,6 +124,16 @@ class AppGroup(click.Group):
         :class:`AppGroup`.
         """
         pass
+
+def _set_app(ctx: click.Context, param: click.Parameter, value: str | None) -> None:
+    pass
+
+def _set_debug(ctx: click.Context, param: click.Parameter, value: bool | None) -> None:
+    pass
+
+def _env_file_callback(ctx: click.Context, param: click.Parameter, value: str | None) -> None:
+    pass
+
 _app_option = click.Option(['-A', '--app'], metavar='IMPORT', help="The Flask application or factory function to load, in the form 'module:name'. Module can be a dotted import or file path. Name is not required if it is 'app', 'application', 'create_app', or 'make_app', and can be 'name(args)' to pass arguments.", is_eager=True, expose_value=False, callback=_set_app)
 _debug_option = click.Option(['--debug/--no-debug'], help='Set debug mode.', expose_value=False, callback=_set_debug)
 _env_file_option = click.Option(['-e', '--env-file'], type=click.Path(exists=True, dir_okay=False), help='Load environment variables from this file. python-dotenv must be installed.', is_eager=True, expose_value=False, callback=_env_file_callback)
