diff --git a/graphene/types/utils.py b/graphene/types/utils.py
index 1c76f7e..ea15348 100644
--- a/graphene/types/utils.py
+++ b/graphene/types/utils.py
@@ -14,6 +14,9 @@ def yank_fields_from_attrs(attrs, _as=None, sort=True):
     """
-    pass
+    return {}
 
+def get_type(_type):
+    pass
+
 def get_underlying_type(_type):
     """Get the underlying type even if it is wrapped in structures like NonNull"""
     pass
\ No newline at end of file
