diff --git a/graphene/utils/props.py b/graphene/utils/props.py
index ec56be0..70c5178 100644
--- a/graphene/utils/props.py
+++ b/graphene/utils/props.py
@@ -3,4 +3,9 @@ class _OldClass:
 
 class _NewClass:
     pass
-_all_vars = set(dir(_OldClass) + dir(_NewClass))
\ No newline at end of file
+_all_vars = set(dir(_OldClass) + dir(_NewClass))
+
+def props(x):
+    return {
+        key: vars(x).get(key, getattr(x, key)) for key in dir(x) if key not in _all_vars
+    }
