diff --git a/graphene/utils/get_unbound_function.py b/graphene/utils/get_unbound_function.py
index e69de29..24f5bf5 100644
--- a/graphene/utils/get_unbound_function.py
+++ b/graphene/utils/get_unbound_function.py
@@ -0,0 +1,2 @@
+def get_unbound_function(func):
+    pass
diff --git a/graphene/utils/module_loading.py b/graphene/utils/module_loading.py
index fd365ca..473dceb 100644
--- a/graphene/utils/module_loading.py
+++ b/graphene/utils/module_loading.py
@@ -9,4 +9,7 @@ def import_string(dotted_path, dotted_attributes=None):
     the first step, and return the corresponding value designated by the
     attribute path. Raise ImportError if the import failed.
     """
-    pass
\ No newline at end of file
+    pass
+
+def lazy_import(dotted_path, dotted_attributes=None):
+    pass
diff --git a/graphene/utils/orderedtype.py b/graphene/utils/orderedtype.py
index 6776272..87315da 100644
--- a/graphene/utils/orderedtype.py
+++ b/graphene/utils/orderedtype.py
@@ -7,6 +7,13 @@ class OrderedType:
     def __init__(self, _creation_counter=None):
         self.creation_counter = _creation_counter or self.gen_counter()
 
+    @staticmethod
+    def gen_counter():
+        pass
+
+    def reset_counter(self):
+        pass
+
     def __eq__(self, other):
         if isinstance(self, type(other)):
             return self.creation_counter == other.creation_counter
diff --git a/graphene/utils/str_converters.py b/graphene/utils/str_converters.py
index 7d0e67b..1646690 100644
--- a/graphene/utils/str_converters.py
+++ b/graphene/utils/str_converters.py
@@ -1 +1,7 @@
-import re
\ No newline at end of file
+import re
+
+def to_camel_case(snake_str):
+    pass
+
+def to_snake_case(name):
+    pass
diff --git a/graphene/utils/trim_docstring.py b/graphene/utils/trim_docstring.py
index ea6a197..d432dd4 100644
--- a/graphene/utils/trim_docstring.py
+++ b/graphene/utils/trim_docstring.py
@@ -1 +1,4 @@
-import inspect
\ No newline at end of file
+import inspect
+
+def trim_docstring(docstring):
+    pass
