diff --git a/graphene/types/scalars.py b/graphene/types/scalars.py
index 7407eb9..2155c39 100644
--- a/graphene/types/scalars.py
+++ b/graphene/types/scalars.py
@@ -31,6 +31,15 @@ class Scalar(UnmountedType, BaseType):
         """
         pass
 MAX_INT = 2147483647
 MIN_INT = -2147483648
+
+def coerce_int(value):
+    pass
+
+def coerce_float(value: Any):
+    pass
+
+def coerce_string(value):
+    pass
 
 class Int(Scalar):
     """
