diff --git a/graphene/types/argument.py b/graphene/types/argument.py
index 3d5d873..24c2e88 100644
--- a/graphene/types/argument.py
+++ b/graphene/types/argument.py
@@ -52,4 +52,7 @@ class Argument(MountedType):
         self.deprecation_reason = deprecation_reason
 
     def __eq__(self, other):
-        return isinstance(other, Argument) and (self.name == other.name and self.type == other.type and (self.default_value == other.default_value) and (self.description == other.description) and (self.deprecation_reason == other.deprecation_reason))
\ No newline at end of file
+        return isinstance(other, Argument) and (self.name == other.name and self.type == other.type and (self.default_value == other.default_value) and (self.description == other.description) and (self.deprecation_reason == other.deprecation_reason))
+
+def to_arguments(args, extra_args=None):
+    pass
diff --git a/graphene/utils/deprecated.py b/graphene/utils/deprecated.py
index a3e7408..c5953fb 100644
--- a/graphene/utils/deprecated.py
+++ b/graphene/utils/deprecated.py
@@ -3,6 +3,9 @@ import inspect
 import warnings
 string_types = (type(b''), type(''))
 
+def warn_deprecation(text):
+    pass
+
 def deprecated(reason):
     """
     This is a decorator which can be used to mark functions
