diff --git a/portalocker/portalocker.py b/portalocker/portalocker.py
index 70217a3..68e5596 100644
--- a/portalocker/portalocker.py
+++ b/portalocker/portalocker.py
@@ -18,4 +18,12 @@ elif os.name == 'posix':
     import fcntl
     LOCKER = fcntl.flock
 else:
-    raise RuntimeError('PortaLocker only defined for nt and posix platforms')
\ No newline at end of file
+    raise RuntimeError('PortaLocker only defined for nt and posix platforms')
+
+
+def lock(file, flags):
+    pass
+
+
+def unlock(file):
+    pass
diff --git a/portalocker/__about__.py b/portalocker/__about__.py
index e16494e..fc965bf 100644
--- a/portalocker/__about__.py
+++ b/portalocker/__about__.py
@@ -6 +6 @@
-__url__ = 'https://github.com/WoLpH/portalocker'
\ No newline at end of file
+__url__ = 'https://github.com/WoLpH/portalocker'
diff --git a/portalocker/constants.py b/portalocker/constants.py
index 51e1148..ec0a47e 100644
--- a/portalocker/constants.py
+++ b/portalocker/constants.py
@@ -38 +38 @@
-    UNBLOCK = LOCK_UN
\ No newline at end of file
+    UNBLOCK = LOCK_UN
diff --git a/portalocker/exceptions.py b/portalocker/exceptions.py
index 27826c9..d2089bf 100644
--- a/portalocker/exceptions.py
+++ b/portalocker/exceptions.py
@@ -17 +17 @@
-    pass
\ No newline at end of file
+    pass
diff --git a/portalocker/redis.py b/portalocker/redis.py
index c6c00ef..e5aa2bd 100644
--- a/portalocker/redis.py
+++ b/portalocker/redis.py
@@ -74 +74 @@
-        self.release()
\ No newline at end of file
+        self.release()
diff --git a/portalocker/utils.py b/portalocker/utils.py
index 3891691..22d05d6 100644
--- a/portalocker/utils.py
+++ b/portalocker/utils.py
@@ -237 +237 @@
-        super().__init__(maximum, name, filename_pattern, directory, timeout, check_interval, fail_when_locked)
\ No newline at end of file
+        super().__init__(maximum, name, filename_pattern, directory, timeout, check_interval, fail_when_locked)
